lib/spack/spack/repo.py

```diff
@@ -962,6 +956,13 @@ def check(condition, msg):
         # Keep name components around for checking prefixes.
         self._names = self.full_namespace.split(".")
 
+        packages_dir = config.get("subdirectory", packages_dir_name)
```

Comment:
If `subdirectory` isn't found would should we allow for a configurable default to be used instead? Or would an `null` subdirectory allow for users to put their `packagex/` directories alongside the repo.yaml file? This came up today in a discussion with @douglasjacobsen.

Reply:
Aren't we changing the schema and documentation to reflect this addition?

Reply:
There is no schema for repo.yaml. I can work on docs for this.